lib/kubernetes-deploy/bindings_parser.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+require 'json'
+require 'csv'
+
+module KubernetesDeploy
+  module BindingsParser
+    extend self
+
+    def parse(string)
+      bindings = parse_json(string) || parse_csv(string)
+
+      unless bindings
+        raise ArgumentError, "Failed to parse bindings."
+      end
+
+      bindings
+    end
+
+    private
+
+    def parse_json(string)
+      bindings = JSON.parse(string)
+
+      unless bindings.is_a?(Hash)
+        raise ArgumentError, "Expected JSON data to be a hash."
+      end
+
+      bindings
+    rescue JSON::ParserError
+      nil
+    end
+
+    def parse_csv(string)
+      lines = CSV.parse(string)
```

Comment:
This might be baseless but I'm not 100% certain that this will parse things that worked before [(which just used `split(',')`)](https://github.com/ruby/ruby/blob/5a8a2706d33849cf42cd56714f519108a4ededfc/lib/optparse.rb#L1924-L1929). The most obvious example would be a newline in a value.

However, the additional quote parsing flexibility would be nice, so maybe the answer is "ship and see"